### DISCUSSION TITLE
Enhancing the Polygon 2D with default options

### DISCUSSION THREAD

**Discussion by jonSP12** [godotengine/godot-proposals#4063] [Editor]
hello,
Is it possible to make some predefined shapes in the sprite polygon 2D
One really cool feature would be the image wrap in photoshop. Being able to make sekleton animations using this type of deform.
Unity already has this shape option by defualt.
![wrap](https://user-images.githubusercontent.com/77213594/154956960-dbba1bb2-1765-41e2-af4b-da50969a493e.png)

![wrap2](https://user-images.githubusercontent.com/77213594/154956981-b30c5442-a96f-48d1-ab69-a8882a2e70a2.png)

**Comment by Xrayez** [comment]
Are you aware of `Sprite` conversion tools? Perhaps could be useful for your case.

![image](https://user-images.githubusercontent.com/17108460/154967169-b306e9cf-148a-48d0-8902-1c609c2053b0.png)

**Reply by jonSP12** [reply]
Iam aware, its showen in the picture above. But i dont know how to deform the sprite in the middle or add extra points to the middle... I was sugesting a pre defined form like photoshop has ( transform image / wrap )

**Reply by Xrayez** [reply]
I recalled now there was a discussion about this:

- https://github.com/godotengine/godot/issues/20601
- https://github.com/godotengine/godot/issues/25120

Looks like it's supported in some way, but perhaps there are no editor tools available?

**Reply by jonSP12** [reply]
that is really good post, its exactly what i was asking... however i dont know maybe it only works by adding a bone or something.
I tried godot 4 alpha, and once i use internal vertices ( points / create internal vertex )...
then connect those vertices ( polygons / create costum polygon ) 
the sprite goes back to its original state with just the size options ( with / height / rotate )

![wrap3](https://user-images.githubusercontent.com/77213594/154996701-3e09ae60-ea95-47aa-9340-1315aae70a08.png)

**Comment by Calinou** [comment]
For reference, this was cross-posted on the Godot forums: https://godotforums.org/discussion/29131/polygon-2d-uv-editor-deform-sprite
